core/system.cpp

```diff
@@ -282,7 +285,54 @@ void Device::device_thread(Device *self)
     }
 }
 
-void Device::send_heartbeat(Device &self)
+std::string System::name(uint8_t component_id)
```

Comment:
Add docs for ALL methods! (ie all the ones above and below here that don't have them!)